### PR TITLE
[Bug] Set upper limit on `dash` temporarily

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
See this for context: https://github.com/plotly/dash/issues/2778

This is supposed to be a temporary solution only such that merging PRs is not blocked. Still waiting for feedback from @l0uden or dash

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/changelog.d/20240304_123956_huong_li_nguyen_add_upper_limit.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+-->
+<!--
+### Added
+
+- A bullet item for the Added category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Changed
```
Comment (maxschulz-COL) [inline vizro-core/changelog.d/20240304_123956_huong_li_nguyen_add_upper_limit.md:26]: we should delete this fragment when this is fixed before the next release

**Review by maxschulz-COL — APPROVED** [review]
Fine by me, we should ideally not release before this is fixed imo

**Review by l0uden — APPROVED** [review]

File: vizro-core/pyproject.toml [hunk 2]
```diff
@@ -15,7 +15,8 @@ classifiers = [
   "Programming Language :: Python :: 3.12"
 ]
 dependencies = [
-  "dash>=2.14.1",  # 2.14.1 needed for compatibility with werkzeug
+  # Temporarily set an upper limit on dash: https://github.com/plotly/dash/issues/2778
+  "dash>=2.14.1, <2.16.0",  # 2.14.1 needed for compatibility with werkzeug
```
[the remark this answers is not in the document]
Comment (huong-li-nguyen) [reply vizro-core/pyproject.toml]: Done

File: vizro-core/changelog.d/20240304_123956_huong_li_nguyen_add_upper_limit.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+- Set upper limit on `dash` temporarily ([#341](https://github.com/mckinsey/vizro/pull/341))
```
Comment (antonymilne) [inline vizro-core/changelog.d/20240304_123956_huong_li_nguyen_add_upper_limit.md:28]:
```suggestion
- Temporarily exclude `dash==2.16.0` ([#341](https://github.com/mckinsey/vizro/pull/341))
```

**Review by antonymilne — APPROVED** [review]

**Pull request merged**